controllers/budget_controller.py

```diff
@@ -1,76 +1,145 @@
 from sqlalchemy.orm import lazyload
 
 from models.Budget import Budget
+from models.Category import Category
 from models.ParentCategory import ParentCategory
 from session import session
+from prettytable import PrettyTable
 
-# GLOBAL VARIABLES
-global_user_id = None
-global_user_name = None
 
+def add_budget(user_id):
+    budget_name = input("Name of the new budget: ")
+    budget = Budget(name=budget_name, user_id=user_id)
+    session.add(budget)
+    session.commit()
+    # new_budget = list(session.query(Budget.id).order_by(Budget.id.desc()).first())[0]
+    new_budget = session.query(Budget.id).order_by(Budget.id.desc()).first()
+    return new_budget.id
 
-# DEFINE LOGIC HERE
-
-# Show user's budget
-def show_budget():
-    # refer to global variables inside function
-    global global_user_id
-    global global_user_name
-
-    # Try to retrieve the user's budget
-    budget_instance = session.query(Budget).filter_by(user_id=global_user_id).first()
-
-    # If user has no budgets
-    if budget_instance is None:
-        print("\nWhoops, you don't have any budgets yet. Shall we create one?")
-        add_budget()
 
-    # else == User does have at least 1 budget
+def select_budget(user_id):
+    budgets = session.query(Budget).filter(Budget.user_id == user_id).all()
+    if len(budgets) == 1:
+        budget_id = budgets[0].id
     else:
-        # Welcome message
-        print("\n{}, here's your budget!".format(global_user_name))
-
-        # Get the list of all user's budgets and its children up to the category level
-        budgets_list = session.query(Budget).filter_by(user_id=global_user_id).options(
-            lazyload(Budget.parent_categories).subqueryload(ParentCategory.categories)).all()
-
-        # Zapytanie wyżej zwraca listę budżetów zalogowanego użytkownika, w której są listy pod-dzieci
-        # Wpisanie budgets_list[0].parent_categories zwraca listę parent kategorii pierwszego budżetu na liście
-        # budgets_list[0].parent_categories[0].categories zwraca listę kategorii i tak dalej
+        budget_id = change_budget(user_id)
+    return budget_id
 
-        # print the name of the first budget on the list
-        print("\n{}".format(budgets_list[0].name.upper()))  # Print first budget's name
 
-        # LOAD WHOLE BUDGET
+def change_budget(user_id):
+    print("\nYour budgets: ")
 
-        # loop through parent categories of the first budget in the list
-        for parent in budgets_list[0].parent_categories:
+    ''' Printing the table with user's budgets.'''
+    # Definition of PrettyTable for budgets.
+    table_show_budgets = PrettyTable()
+    table_show_budgets.field_names = ["id", "name"]
 
-            # Start calculating the parent's available amount based its children' available_amount
-            parent_available_sum = 0.00
+    # Reading data from databe and inserting into PrettyTable
+    for instance in session.query(Budget).filter(Budget.user_id == user_id):
```

Comment:
Tego typu kwerki można zapisywać tak:

`for budget_instance in user.budgets:`

Starajmy się nie dawać suchego "instance", bo przy bardziej skpmplikowanej logice nie będzie wiadomo, do jakiego obiektu się odnosimy

O wiele lepiej jest pisać "user_instance" "budget_instance" etc.

Reply:
O, to mamy do poprawy kilka kwerek

Reply:
Poprawione